iroh/src/magicsock/remote_map/remote_state.rs

```diff
@@ -688,7 +726,9 @@ impl RemoteStateActor {
                 });
             }
             Err(err) => {
+                // TODO: distinguish between fatal and temporary errors
                 warn!("failed to initiate NAT traversal: {err:#}");
+                self.needs_holepunching.set(true).ok();
```

Comment:
Doesn't this mechanism already exist with `self.scheduled_holepunch`? Can we not re-use that?